includes/constants/class-intent-status.php

```diff
@@ -26,4 +26,15 @@ class Intent_Status extends Base_Constant {
 	const REQUIRES_CAPTURE        = 'requires_capture';
 	const CANCELED                = 'canceled';
 	const SUCCEEDED               = 'succeeded';
+
+	/**
+	 * Stripe intents that are treated as successfully created.
+	 *
+	 * @type array
+	 */
+	const SUCCESSFUL_STATUSES = [
```

Comment:
@htdat now that we have the abstract intention, what are your thoughts on adding an `is_successful()` method to it, and using it to check the status there? By doing so, we would not need this constant, and less code would be repeated.

Reply:
@RadoslavGeorgiev - good idea, however, I think, we might think more about the method name. `is_successful()` can be confusing with "this intent status is `succeeded`". What if we come with something like `is_authorized`? Otherwise, the plural noun `statuses` forces other developers to double check what's they are.

Reply:
Yes, that makes sense to me. 👍

Reply:
Fixed 2a3d46285c3744ba9440248394da0bcdb32886bb